packages/web/src/components/apps/sidebar.tsx

```diff
@@ -43,7 +44,25 @@ export default function Sidebar() {
   const [showFeedback, setShowFeedback] = useState(false);
   const [panel, _setPanel] = useState<PanelType | null>(openedFile === null ? 'explorer' : null);
 
+  // When the user changes to the preview, close the open side panel
+  // When the user goes back to the code, re-open the side panel if it was already opened
+  const { tab, switchTab } = useHeaderTab();
+  const [sidePanelPreviousValue, setSidePanelPreviousValue] = useState<PanelType | null>(null);
+  useEffect(() => {
+    if (tab === 'code' && sidePanelPreviousValue !== null) {
+      _setPanel(sidePanelPreviousValue);
+      setSidePanelPreviousValue(null);
+      return;
+    }
+
+    if (tab === 'preview' && panel !== null) {
+      setSidePanelPreviousValue(panel);
+      _setPanel(null);
+    }
+  }, [tab, panel, sidePanelPreviousValue]);
+
   function setPanel(nextPanel: PanelType) {
+    switchTab('code');
     _setPanel(nextPanel === panel ? null : nextPanel);
   }
```

Comment:
This behavior, through experimentation, seemed like it made sense. Effectively reopen the previously opened sidebar panel when a user goes back to "code" from "preview".